test/parallel.bats

```diff
@@ -208,3 +211,10 @@ check_parallel_tests() { # <expected maximum parallelity>
 @test "BATS_NO_PARALLELIZE_WITHIN_FILE does not work from inside test function" {
   DISABLE_IN_TEST_FUNCTION=1 reentrant_run ! bats --jobs 2 "$FIXTURE_ROOT/must_not_parallelize_within_file.bats"
 }
+
+@test "Short form typo does not run endlessly" {
+  unset BATS_NO_PARALLELIZE_ACROSS_FILES
+  run bats -j2 "$FIXTURE_ROOT/../bats/passing.bats"
```

Comment:
If you really want to test the check that was added in this PR then I think it should be:

```suggestion
  run bats -j -2 "$FIXTURE_ROOT/../bats/passing.bats"
```

Otherwise this test will switch to covering a completely different part of the code the moment you refactor the options parser.

Reply:
Additional test submitted in #693